components/admin_console/team_channel_settings/team/details/team_details.jsx

```diff
@@ -330,19 +358,109 @@ export default class TeamDetails extends React.PureComponent {
     hideRemoveUsersModal = () => {
         this.setState({showRemoveConfirmation: false});
     }
-    showRemoveUsersModal = () => {
-        if (this.state.usersToRemoveCount > 0) {
+
+    hideArchiveConfirmModal = () => {
+        this.setState({showArchiveConfirmModal: false});
+    };
+
+    onSave = () => {
+        if (this.teamToBeArchived()) {
+            this.setState({showArchiveConfirmModal: true});
+        } else if (this.state.usersToRemoveCount > 0) {
             this.setState({showRemoveConfirmation: true});
         } else {
             this.handleSubmit();
         }
     }
 
+    teamToBeArchived = () => {
+        const {isLocalArchived} = this.state;
+        const isServerArchived = this.props.team.delete_at !== 0;
+        return isLocalArchived && !isServerArchived;
+    }
+
+    teamToBeRestored = () => {
+        const {isLocalArchived} = this.state;
+        const isServerArchived = this.props.team.delete_at !== 0;
+        return !isLocalArchived && isServerArchived;
+    }
+
+    onToggleArchive = () => {
+        const {isLocalArchived, serverError, previousServerError} = this.state;
+        const {isDisabled} = this.props;
+        if (isDisabled) {
+            return;
+        }
+        const newState = {
+            saveNeeded: true,
+            isLocalArchived: !isLocalArchived,
+        };
+
+        if (newState.isLocalArchived) {
+            // if the channel is being archived then clear the other server
+            // errors, they're no longer relevant.
+            newState.previousServerError = serverError;
+            newState.serverError = null;
+        } else {
+            // if the channel is being unarchived (maybe the user had toggled
+            // and untoggled) the button, so reinstate any server errors that
+            // were present.
+            newState.serverError = previousServerError;
+            newState.previousServerError = null;
+        }
```

Comment:
I'm not sure that I understand this logic. Why is toggling the archive button showing and hiding a server error? Is it possible that the error has nothing to do with archiving? I'm thinking we should show the error until we have another form submission.

Reply:
Suppose there is a server error that is because of removing or adding a user in the team, but now the user is archiving the team so those server errors are irrelevant now and if the user has just toggled and untoggled the archive button then those errors should persist in the state. Also, on the channel details screen, the server errors are handled in the same way on archiving/unarchiving the channel. That's why I added the same logic here.